src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -180,6 +190,28 @@ def _add_whl_ext(cli_ctx, source, ext_sha256=None, pip_extra_index_urls=None, pi
     return extension_name
 
 
+def _check_setup_extras(install_setup_extras, ext_file, extension_name):
+    # Check the setup extras keywords
+    extras = ""
+    if install_setup_extras:
+        extras = WheelExtension.get_metadata_extras(ext_file)
+        if not extras:
+            raise CLIError(
+                "The specified version of extension '{}' does not support setup "
+                "extras.".format(extension_name)
+            )
+        for ext in install_setup_extras:
+            if ext not in extras:
+                raise CLIError(
+                    "The setup extras keyword {} is invalid. The extension {} currently supports "
+                    "these keywords: {}.".format(ext, extension_name, extras)
+                )
+
+        extras = "[{}]".format(",".join(install_setup_extras))
```

Comment:
This `_check_setup_extras` does more than "check". It actually returns the extras in [PEP 508 convention](https://peps.python.org/pep-0508/#extras), like `requests[security]`.

Reply:
renamed